src/App.jsx

```diff
@@ -14,33 +16,90 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
-export const App = () => (
-  <div className="section content">
-    <div className="buttons">
-      <button type="button" className="button is-info is-light">
-        Sort alphabetically
-      </button>
+const SORT_BY_ALPHABET = 'Sort alphabetically';
+const SORT_BY_LENGTH = 'Sort by length';
 
-      <button type="button" className="button is-success is-light">
-        Sort by length
-      </button>
+export const App = () => {
+  const [visibleGoods, setVisibleGoods] = useState(goodsFromServer);
+  const [sort, setSort] = useState({ type: '', reversed: false });
 
-      <button type="button" className="button is-warning is-light">
-        Reverse
-      </button>
+  // function for sort by length and alphabet
+  const handleSort = type => {
+    const sortedGoods = [...goodsFromServer];
 
-      <button type="button" className="button is-danger is-light">
-        Reset
-      </button>
-    </div>
+    if (type === SORT_BY_ALPHABET) {
+      sortedGoods.sort((a, b) => a.localeCompare(b));
+    } else if (type === SORT_BY_LENGTH) {
+      sortedGoods.sort((a, b) => a.length - b.length);
+    }
+
+    if (sort.reversed) {
+      sortedGoods.reverse();
+    }
+
+    setSort({ type, reversed: sort.reversed });
```

Comment:
- In your handlers just set state. Sorting logic move to another function(and as I mentioned move it outside the component, like it's a helper function) that will sort and reverse goods if need.
- When you updating state and want to store the previous one use functional setState
`setSortParams(prev => ({ ...prev, type });`